### ISSUE TITLE
Feature request: Support decoding SendGrid URLs

### ISSUE THREAD

**Issue by cfroystad** [sendgrid/hacktoberfest#26]
### TLDR;

An API endpoint for decoding SendGrid tracking URLs would allow security systems to inspect URLs in emails without revealing to the sender that the email recipient has an operational email inbox.

### Background

While SendGrid seems to be doing a good job at combating spammers and phishing, we're still seeing quite a few phishing campaigns using SendGrid as their method of delivery. The use of SendGrid tracking URLs makes it difficult to inspect the included URLs without revealing to the sender that the email address is operational.

### Feature request

An API endpoint where we could submit a list of SendGrid URLs and get back their real URL.

_In an ideal world, it would be great to be able to decode the URL without having to ask your API (e.g. like reddit link tracking), but due to backwards compatibility, I guess that's unrealistic for now._

**Comment by DirkPryde** [comment]
Something like this would be amazing.
I have been looking for ways to decode SendGrid URLs (even if I could find the logic to decode myself) as more and more of our users are reporting SendGrid URLs because they don't look safe.

**Comment by cfroystad** [comment]
After some thinking, I found an annoyingly simple solution 🙂 

For anyone else in need of this functionality, it's kind of already supported. You'll have to send a GET request to the SendGrid URL with a client in which you disallow redirects. You'll get a `Location` header back which is the URL to which the SendGrid URL would otherwise have redirected your request.
